WcaOnRails/lib/advancement_conditions/advancement_condition.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module AdvancementConditions
+  class AdvancementCondition
+    include ActiveModel::Validations
+
+    attr_accessor :level
+    validates :level, numericality: { only_integer: true }
+
+    @@advancement_conditions = [AttemptResultCondition, PercentCondition, RankingCondition].freeze
```

Comment:
This is a change compared to the initial file, where `AdvancementCondition.subclasses` was used, but when lazy loading is used by spring the file containing the subclasses are not loaded at this point.
In any case it's more explicit to state exactly which advancement conditions we want available.